src/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -141,6 +141,10 @@ public static MemoryMappedFile CreateFromFile(string path, FileMode mode, string
             {
                 throw new ArgumentException(SR.Argument_NewMMFAppendModeNotAllowed, nameof(mode));
             }
+            if (mode == FileMode.Truncate)
+            {
+                throw new ArgumentException(SR.Argument_NewMMFTruncateModeNotAllowed, nameof(mode));
```

Comment:
I think it would be wise to disable FileMode.Truncate to match netfx. Unintentional changes like allowing FileMode.Truncate where we previously always threw make me nervous. If we want to stay with the netcoreapp behavior then we should audit our tests and really make sure there aren't any unintentional breaks that we missed.